[Bug]: IOException when exporting project to Zip inside ${PROJECT_HOME}

### Apache Hop version?

2.9.0

### Java version?

21.0.4

### Operating system

Windows

### What happened?

We face 2 issues when trying to export the current project to zip. This bug is about the second issue.

Steps to reproduce:
1. Open / Create a project
2. Click on HOP icon for menu actions
3. Click 'Export current project to zip'
4. Chose a valid destination file INSIDE PROJECT_HOME! (In the file selection dialog, the first issue appears. Its not exclusive to this workflow and depends on the current active pipeline/workflow in the view, therefore I'll skip its description here. Sum up: The default file path provided can be invalid.)
5. Press 'Save' and see error message:

```
java.lang.reflect.InvocationTargetException: Error zipping project: 

java.io.IOException: Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!
Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!

Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!

	at org.apache.hop.projects.gui.ProjectsGuiPlugin.lambda$menuProjectExport$1(ProjectsGuiPlugin.java:1152)
	at org.apache.hop.ui.core.dialog.ProgressMonitorDialog.lambda$run$2(ProgressMonitorDialog.java:166)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.apache.hop.core.exception.HopFileException: 

java.io.IOException: Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!
Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!

Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!

	at org.apache.hop.core.vfs.HopVfs.getOutputStream(HopVfs.java:326)
	at org.apache.hop.projects.gui.ProjectsGuiPlugin.lambda$menuProjectExport$1(ProjectsGuiPlugin.java:1137)
	... 2 more
Caused by: java.io.IOException: Parent directory file:///C:/Users/DennisvanderWals/hop/${PROJECT_HOME} does not exist!
	at org.apache.hop.core.vfs.HopVfs.getOutputStream(HopVfs.java:288)
	at org.apache.hop.core.vfs.HopVfs.getOutputStream(HopVfs.java:324)
	... 3 more
```

Instead of resolving ${PROJECT_HOME}, a path inside the installation directory of hop containing ${PROJECT_HOME} is tried to be opened.
Zipping is successful at other directories than inside project home.

### Issue Priority

Priority: 2

### Issue Component

Component: Hop Gui

--- a/plugins/misc/projects/src/main/java/org/apache/hop/projects/gui/ProjectsGuiPlugin.java
+++ b/plugins/misc/projects/src/main/java/org/apache/hop/projects/gui/ProjectsGuiPlugin.java
@@ -1073,14 +1073,17 @@ public class ProjectsGuiPlugin {
   public void menuProjectExport() {
     HopGui hopGui = HopGui.getInstance();
     Shell shell = hopGui.getShell();
+    IVariables variables = hopGui.getVariables();
 
+    // Resolve variables in filepath
     String zipFilename =
-        BaseDialog.presentFileDialog(
-            true,
-            shell,
-            new String[] {"*.zip", "*.*"},
-            new String[] {"Zip files (*.zip)", "All Files (*.*)"},
-            true);
+        variables.resolve(
+            BaseDialog.presentFileDialog(
+                true,
+                shell,
+                new String[] {"*.zip", "*.*"},
+                new String[] {"Zip files (*.zip)", "All Files (*.*)"},
+                true));
     if (zipFilename == null) {
       return;
     }
@@ -1104,7 +1107,6 @@ public class ProjectsGuiPlugin {
               monitor.setTaskName(
                   BaseMessages.getString(PKG, "ProjectGuiPlugin.ZipDirectory.Taskname.Text"));
               HashMap variablesMap = new HashMap<>();
-              IVariables variables = hopGui.getVariables();
 
               for (String name : variables.getVariableNames()) {
                 if (!name.contains("java.")
@@ -1144,7 +1146,11 @@ public class ProjectsGuiPlugin {
               String projectHomeFolder =
                   HopVfs.getFileObject(projectHome).getParent().getName().getURI();
               zipFile(
-                  projectDirectory, projectDirectory.getName().getURI(), zos, projectHomeFolder);
+                  projectDirectory,
+                  projectDirectory.getName().getURI(),
+                  zos,
+                  projectHomeFolder,
+                  zipFilename);
               zipFile(
                   HopVfs.getFileObject(
                       Const.HOP_CONFIG_FOLDER + Const.FILE_SEPARATOR + Const.HOP_CONFIG),
@@ -1152,7 +1158,8 @@ public class ProjectsGuiPlugin {
                       + Const.FILE_SEPARATOR
                       + Const.HOP_CONFIG,
                   zos,
-                  projectHomeFolder);
+                  projectHomeFolder,
+                  zipFilename);
               zipString(
                   variablesJson, "variables.json", zos, projectDirectory.getName().getBaseName());
               zipString(
@@ -1190,7 +1197,8 @@ public class ProjectsGuiPlugin {
       FileObject fileToZip,
       String filename,
       ZipOutputStream zipOutputStream,
-      String projectHomeParent)
+      String projectHomeParent,
+      String zipFilename)
       throws IOException {
     if (fileToZip.isHidden()) {
       return;
@@ -1206,11 +1214,15 @@ public class ProjectsGuiPlugin {
       }
       FileObject[] children = fileToZip.getChildren();
       for (FileObject childFile : children) {
+        if (childFile.equals(zipFilename)) {
+          return;
+        }
         zipFile(
             childFile,
             filename + "/" + childFile.getName().getBaseName(),
             zipOutputStream,
-            projectHomeParent);
+            projectHomeParent,
+            zipFilename);
       }
       return;
     }

